packages/components/src/data-model/word-query/word-query-resolver.js

```diff
@@ -0,0 +1,82 @@
+import { makeVar } from '@apollo/client/core'
+import { Logger, Digest } from 'alpheios-data-models'
+import WordQuery from '@comp/data-model/word-query/word-query.js'
+import WordQueryResult from '@comp/data-model/word-query/word-query-result.js'
+
+/**
+ * A class that would receive GraphQL requests and resolve them with data.
+ * Creates an instance of WordQuery for the query resolution.
+ */
+export default class WordQueryResolver {
+  constructor ({ getLexiconsFn, getShortLexiconsFn } = {}) {
+    this._getLexiconsFn = getLexiconsFn
+    this._getShortLexiconsFn = getShortLexiconsFn
+  }
+
+  // This function will be called by Apollo when new GraphQL query will arrive.
+  resolve (_, { storage, variables }) {
+    // This is a unique key of each request. We'll use its hash value to cache requests already resolved.
+    const key = Digest.fromObject({
+      word: variables.word,
+      language: variables.language,
+      contextForward: variables.contextForward,
+      contextBackward: variables.contextBackward
+    })
+    // The words map will store references to the results stored in the Apollo cache
+    if (!storage.words) { storage.words = new Map() }
+    if (!storage.words.has(key)) {
+      const wqResultWatcher = {
+        // This function will track changes in the top-level props of the result object
+        set: function (obj, prop, value) {
+          let wordReactive = storage.words.get(key) // eslint-disable-line prefer-const
+          let word = wordReactive() // eslint-disable-line prefer-const
+          if (prop === 'homonymGroup') {
+            // Update the value of the reactive variable
+            const homonyms = value.toJsonObject().homonyms
+            // Need to recreate an object on order for reactive var to be updated
+            wordReactive({ ...word, homonyms })
+          } else if (prop === 'state') {
+            // Update the value of the reactive variable
+            const state = WordQueryResult.stateToJsonObject(value)
+            // Need to recreate an object on order for reactive var to be updated
+            wordReactive({ ...word, state })
+          } else if (prop === 'errors') {
+            // Update the value of the reactive variable
+            const errors = WordQueryResult.errorsToJsonObject(value)
+            // Need to recreate an object on order for reactive var to be updated
+            wordReactive({ ...word, errors })
```

Comment:
The reason for the difference is that in case of a `homonymGroup` prop update the `value` is the "real" `HomonymGroup` object with methods while `state` and `errors` are JSON object representations with no methods on them. I think it's a good suggestion to use static methods in all the cases and I will update the code accordingly.